client/categories.py

```diff
@@ -0,0 +1,113 @@
+        'groups:posts:view_deleted',
+        'groups:purge',
+        'groups:moderate'
+    ]
+
+    default_groups = [
+        'registered-users',
+        'guests',
+        'spiders'
+    ]
+
+    def __init__(self):
+        super(NodeBBCategory, self).__init__()
+
+    def create(self, course_id, **payload):
+        """
+        Creates a new NodeBB Category.
+
+        Args:
+            course_id (Course Key) : Course_Key
+
+            **payload (Dictionary): All other params to pass for category creation, like category_name, type.
+
+        Returns:
+            tuple: Tuple in the form (response_code, json_response) received from requests call.
+
+        """
+        response_code, json_response = self.post('/api/v2/categories', **payload)
+
+        if response_code == 200:
+            save_category_relation_into_db(course_id=course_id, category_id=json_response['cid'])
+
+        return response_code, json_response
+
+    def delete_default_permissions(self, category_id):
+        """
+        By default whenever a new category is created in NodeBB, the default groups have some default privileges
+        of that category. So, first deletes those default privileges in order to assign some specific privileges
+        to the groups that are related to that category.
+
+        Args:
+            category_id (int): Category from which we are going to delete privileges.
+
+        Returns:
+             tuple: Tuple in the form (response_code, json_response) received from requests call.
+        """
+        payload = {
+            'privileges': self.default_privileges,
+            'groups': self.default_groups
+        }
+        response_code, json_response = self.delete('/api/v2/categories/{}/privileges'.format(category_id), **payload)
+        return response_code, json_response
+
+    def add_course_group_permission(self, category_id, group_name):
+        """
+        Assigns group and privileges to category.
+
+        Args:
+            category_id (int): Category to assign group and privileges.
+            group_name (str): Group to assign to Category.
+
+        Returns:
+            tuple: Tuple in the form (response_code, json_response) received from requests call.
+        """
+        payload = {
+            'privileges': self.default_privileges,
+            'groups': [
+                group_name
+            ]
+        }
+        response_code, json_response = self.put('/api/v2/categories/{}/privileges'.format(category_id), **payload)
+        return response_code, json_response
+
+    def delete_category(self, category_id):
+        """
+        Assigns group and privileges to category.
+
+        Args:
+            category_id (int): Category to delete.
+
+        Returns:
+            tuple: Tuple in the form (response_code, json_response) received from requests call.
+        """
+        response_code, json_response = self.delete('/api/v2/categories/{}'.format(category_id))
```

Comment:
the variables are unnecessary, You can simply write
```
return self.delete('/api/v2/categories/{}'.format(category_id))
```

Reply:
We are using these variables only for making our code more descriptive.